### ISSUE TITLE
How are array jobs treated. 

### ISSUE THREAD

**Issue by Sabryr** [NAICNO/Jobanalyzer#675]
- Here is an example on fox: ArrayJobId=969056
- The user is running on 20 nodes. 
How to check the resource usage as a cumulative analysis using the array jobid  (without using each job id)

**Comment by lars-t-hansen** [comment]
Yeah, array jobs are not well supported at the moment.  It requires joining the sonar data with the slurm data (or possibly it requires sonar observing more cgroup slurm info).  It can be done but there's no UX for it, and what little support there is for it in Jobanalyzer wrt the slurm data is likely buggy.

**Comment by lars-t-hansen** [comment]
Relevant: #584 #676 .

**Comment by lars-t-hansen** [comment]
Sketching:

- the `sonalyze jobs` output can have a field `ParentJob` which is zero normally but is the job ID of the uppermost job for jobs that are part of array jobs or het jobs, that will give consumers something to work with
- ditto the report could have a boolean, `IsParent` if the job is such a parent job (cheap to compute i think)
- listing the subjobs could be a filter, `-parent-job J`, just like there is now a `-job`: this would select jobs whose parent is J

The alternative is to let `sonalyze jobs` be only about sonar data and `sonalyze sacct` be about slurm data, and to let the client do the lookups and joining.  The latter will want some more filtering options and better array/het support but it wants that anyway.  In that case, we should write a cookbook / some sample code about how to do the joining properly.
